README
How can I generate the inline documentation of date2cal?

sudo pear install PhpDocumentor
phpdoc -d ./src/ -t doc/phpdoc/ -ue -ti Date2Calendar -s


ChangeLog
19-01-10 Stefan Galinski  <stefan.galinski@gmail.com>
	* 7.3.2: fixed some issues inside the natural language parser helpfiles
	* 7.3.2: fixed wrong english error message
	* 7.3.2: added the current wanted format as a first pattern in all languages, because the current format should always be available! (#5976)
	* 7.3.2: added +2, -3 days parser patterns (#2391)
	* 7.3.2: increaded help window height to 550px

12-01-10 Stefan Galinski  <stefan.galinski@gmail.com>
	RELEASE OF VERSION 7.3.1
	* 7.3.1: added creation information of the typo3temp directory tx_date2cal to the ext_emconf.php file
	* 7.3.1: updated copyright year
	* 7.3.1: added missing xclass inclusion codes
	* 7.3.1: updated the manual to the new format
	* 7.3.1: added swedish language files (Thanks to Björn T. Hallberg)
	* 7.3.1: added slovak language files (Thanks to Peter Žák)

10-12-09 Stefan Galinski  <stefan.galinski@gmail.com>
	RELEASE OF VERSION 7.3.0
	* 7.3.0: added 4.3 compatibility
	* 7.3.0: introduced option to disable the backend part of the extension
	* 7.3.0: added backend compatibility to the natural language parser mode
	* 7.3.0: changed the renderImages method to act as a wrapper for the combined calls to the renderCalendarImages and getConfigJS methods (deprecated method!)
	* 7.3.0: some code cleanups and fixes to the code
	* 7.3.0: added support of the daFormat in the API for the natural language parser
	* 7.3.0: updated documentation

06-12-09 Stefan Galinski  <stefan.galinski@gmail.com>
	* 7.3.0: improved the localisation/internationalisation handling of the natural language parser
	* 7.3.0: added german localisation and internationalisation for the nlp mode
	* 7.3.0: removed the size and maxlength restrictions of the input field (for easier styling in the frontend)
	* 7.3.0: fixed vertical alignment of the calendar inside the backend
	* 7.3.0: default input field id changed to a random hash based upon the microtime (before: static string "date")
	* 7.3.0: fixed parser warning, if a month or weekday is ambigous
	* 7.3.0: localizable error messages
	* 7.3.0: cleanup of the existing natural language parser patterns
	* 7.3.0: added new feature to inject all known attributes to each of the rendered html tags
	* 7.3.0: fixed that the checkbox isn't checked if a value was given
	* 7.3.0: checkbox javascript is added directly into the html code now
	* 7.3.0: fixed firing of the onchange event handler after adding a date with the calendar
	* 7.3.0: added XCLASS code for the API class
	* 7.3.0: fixed issue with empty configuration parameters in the API class

05-12-09 Stefan Galinski  <stefan.galinski@gmail.com>
	* 7.3.0: renamed folder res to resources
	* 7.3.0: added folder images inside resources and moved the image files
	* 7.3.0: moved jscalendar folder to resources
	* 7.3.0: added naturalLanguageParser folder to resources and moved the related file into it
	* 7.3.0: moved javascript folder to resources
	* 7.3.0: renamed options helpImg and calImg to helpIcon and calendarIcon
	* 7.3.0: moved temporary cache folder to typo3temp/tx_date2cal/
	* 7.3.0: added l10n/i18n to the naturalLanguageParser
	* 7.3.0: increased the version requirements and added PHP 5.3 and TYPO3 4.3 as compatible versions
	* 7.3.0: improved the API documentation
	* 7.3.0: added access modifiers (bye bye PHP4) and renamed some internal methods/properties

03-12-09 Stefan Galinski  <stefan.galinski@gmail.com>
	* 7.3.0: rewrite of the natural language parser (removed the error-prone prototype dependency)!

29-11-09 Stefan Galinski  <stefan.galinski@gmail.com>
	* 7.3.0: absolute path to the help page to fix frontends with absRefPrefix
	* 7.3.0: modified extension description
	* 7.3.0: removed inline documentation
	* 7.3.0: removed netbeans stuff

13-11-08 Stefan Galinski  <stefan.galinski@gmail.com>
	RELEASE OF VERSION 7.2.0
	* 7.2.0: fixed API bug (prototype needs to be included before the natural language parser)
	* 7.2.0: updated the manual

12-11-08 Stefan Galinski  <stefan.galinski@gmail.com>
	* 7.2.0: added function to get calendar option to the jscalendar API
	* 7.2.0: added support of yyyy.mm.dd and yy.mm.dd formats
	* 7.2.0: added fixed formats to the backend calendar again (d-m-Y, m-d-Y)

11-11-08 Stefan Galinski  <stefan.galinski@gmail.com>
	* 7.2.0: added support of TYPO3_CONF_VARS['sys']['ddmmyy']
	* 7.2.0: added support of the USdateFormat option in relation with ddmmyy option
	* 7.2.0: added support of short years (%y) to the natural language parser
	* 7.2.0: added support of own date formats in the jscalendar API

10-11-08 Stefan Galinski  <stefan.galinski@gmail.com>
	RELEASE OF VERSION 7.1.1
	* 7.1.1: fixed "%d.%m.%Y" and "%m/%d/%Y" date formats
	* 7.1.1: more stable calendar css code
	* 7.1.1: fixed php error with date2cal in frontend and no backend cookie (#1955, Thanks to Robert Heel and Fabrizio Branca)

29-10-08 Stefan Galinski  <stefan.galinski@gmail.com>
	* 7.1.1: added option to disable the inclusion of prototype
	* 7.1.1: refactoring of the script generation function of the jscalendar class
	* 7.1.1: fixed problems with the option "absRefPrefix"
	* 7.1.1: fixed overriding of local user/group preferences (Thanks to Mickaël Perrin)
	* 7.1.1: fixed some issues in the manual
	* 7.1.1: updated the doxyfile

28-10-08 Stefan Galinski  <stefan.galinski@gmail.com>
	* 7.1.1: php code cleanup to fit the TYPO3 CGL
	* 7.1.1: fixed warning in ext_tables.php if a TCA table has an empty columns section
	* 7.1.1: fixed an issue with duplicate input id generation with flexforms (md5ID is used now)
	* 7.1.1: deactivate secOptionsAlwaysOn (not needed anymore) and datetime (never worked) option by default
	* 7.1.1: added improved french translation (Thanks to Christophe Marcoccia)
	* 7.1.1: renamed the chinese language file to get it working
	* 7.1.1: updated manual

27-10-08 Stefan Galinski  <stefan.galinski@gmail.com>
	* 7.1.1: added kj_becalendar to the ignore list (Thanks to Thomas Kofler)

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext: "date2cal"
#
# Auto generated 14-01-2010 16:07
#
# Manual updates:
# Only the data in the array - anything else is removed by next write.
# "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Date2Calendar',
	'description' => 'Extends all backend date/datetime fields with a calendar in relation with a natural language parser. The usage in other extension is supported by an API.',
	'category' => 'be',
	'shy' => 0,
	'version' => '7.3.1',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => 'bottom',
	'loadOrder' => '',
	'module' => '',
	'state' => 'stable',
	'uploadfolder' => 0,
	'createDirs' => 'typo3temp/tx_date2cal',
	'modify_tables' => '',
	'clearcacheonload' => 0,
	'lockType' => '',
	'author' => 'Stefan Galinski',
	'author_email' => 'stefan.galinski@gmail.com',
	'author_company' => '',
	'CGLcompliance' => '',
	'CGLcompliance_note' => '',
	'constraints' => array(
		'depends' => array(
			'php' => '5.0.0-5.3.99',
			'typo3' => '4.0.0-4.3.99',
		),
		'conflicts' => array(
			'erotea_date2cal' => '',
			'kj_becalendar' => '',
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:84:{s:9:"ChangeLog";s:4:"8786";s:16:"de.locallang.xml";s:4:"1aca";s:21:"ext_conf_template.txt";s:4:"724f";s:12:"ext_icon.gif";s:4:"78f7";s:17:"ext_localconf.php";s:4:"0b0a";s:14:"ext_tables.php";s:4:"c1ba";s:13:"locallang.xml";s:4:"9c8d";s:29:"resources/images/calendar.png";s:4:"5307";s:30:"resources/images/calendar2.gif";s:4:"c1e5";s:29:"resources/images/helpIcon.png";s:4:"b079";s:30:"resources/jscalendar/ChangeLog";s:4:"5628";s:38:"resources/jscalendar/calendar-setup.js";s:4:"cf54";s:32:"resources/jscalendar/calendar.js";s:4:"e1d0";s:40:"resources/jscalendar/skins/menuarrow.gif";s:4:"1f8c";s:41:"resources/jscalendar/skins/menuarrow2.gif";s:4:"1f8c";s:45:"resources/jscalendar/skins/aqua/active-bg.gif";s:4:"f8fb";s:43:"resources/jscalendar/skins/aqua/dark-bg.gif";s:4:"949f";s:44:"resources/jscalendar/skins/aqua/hover-bg.gif";s:4:"803a";s:45:"resources/jscalendar/skins/aqua/menuarrow.gif";s:4:"1f8c";s:45:"resources/jscalendar/skins/aqua/normal-bg.gif";s:4:"8511";s:47:"resources/jscalendar/skins/aqua/rowhover-bg.gif";s:4:"c097";s:45:"resources/jscalendar/skins/aqua/status-bg.gif";s:4:"1238";s:41:"resources/jscalendar/skins/aqua/theme.css";s:4:"4877";s:44:"resources/jscalendar/skins/aqua/title-bg.gif";s:4:"8d65";s:44:"resources/jscalendar/skins/aqua/today-bg.gif";s:4:"9bef";s:47:"resources/jscalendar/skins/skin_grey2/theme.css";s:4:"7f96";s:43:"resources/jscalendar/skins/t3skin/theme.css";s:4:"07c4";s:44:"resources/jscalendar/skins/t3skin2/theme.css";s:4:"98f9";s:40:"resources/jscalendar/lang/calendar-af.js";s:4:"8a39";s:40:"resources/jscalendar/lang/calendar-al.js";s:4:"5a46";s:40:"resources/jscalendar/lang/calendar-bg.js";s:4:"eaa5";s:47:"resources/jscalendar/lang/calendar-big5-utf8.js";s:4:"9eaa";s:42:"resources/jscalendar/lang/calendar-big5.js";s:4:"d4f2";s:40:"resources/jscalendar/lang/calendar-br.js";s:4:"f4c3";s:40:"resources/jscalendar/lang/calendar-ca.js";s:4:"a912";s:45:"resources/jscalendar/lang/calendar-cn-utf8.js";s:4:"0a98";s:45:"resources/jscalendar/lang/calendar-cs-utf8.js";s:4:"6119";s:44:"resources/jscalendar/lang/calendar-cs-win.js";s:4:"1776";s:40:"resources/jscalendar/lang/calendar-da.js";s:4:"c942";s:40:"resources/jscalendar/lang/calendar-de.js";s:4:"f811";s:40:"resources/jscalendar/lang/calendar-du.js";s:4:"5c26";s:40:"resources/jscalendar/lang/calendar-el.js";s:4:"b2b4";s:40:"resources/jscalendar/lang/calendar-en.js";s:4:"4681";s:40:"resources/jscalendar/lang/calendar-es.js";s:4:"9760";s:40:"resources/jscalendar/lang/calendar-fi.js";s:4:"f385";s:45:"resources/jscalendar/lang/calendar-fr-utf8.js";s:4:"49ae";s:45:"resources/jscalendar/lang/calendar-he-utf8.js";s:4:"4f95";s:45:"resources/jscalendar/lang/calendar-hr-utf8.js";s:4:"7d22";s:40:"resources/jscalendar/lang/calendar-hr.js";s:4:"e83c";s:40:"resources/jscalendar/lang/calendar-hu.js";s:4:"a884";s:40:"resources/jscalendar/lang/calendar-it.js";s:4:"4e7a";s:40:"resources/jscalendar/lang/calendar-jp.js";s:4:"9071";s:45:"resources/jscalendar/lang/calendar-ko-utf8.js";s:4:"ea3e";s:40:"resources/jscalendar/lang/calendar-ko.js";s:4:"e0e3";s:45:"resources/jscalendar/lang/calendar-lt-utf8.js";s:4:"cfac";s:40:"resources/jscalendar/lang/calendar-lt.js";s:4:"0175";s:40:"resources/jscalendar/lang/calendar-lv.js";s:4:"77e7";s:40:"resources/jscalendar/lang/calendar-nl.js";s:4:"01a6";s:40:"resources/jscalendar/lang/calendar-no.js";s:4:"931f";s:45:"resources/jscalendar/lang/calendar-pl-utf8.js";s:4:"f92e";s:40:"resources/jscalendar/lang/calendar-pl.js";s:4:"d4d0";s:40:"resources/jscalendar/lang/calendar-pt.js";s:4:"7d8b";s:40:"resources/jscalendar/lang/calendar-ro.js";s:4:"559f";s:40:"resources/jscalendar/lang/calendar-ru.js";s:4:"bc19";s:45:"resources/jscalendar/lang/calendar-ru_win_.js";s:4:"b1fa";s:40:"resources/jscalendar/lang/calendar-si.js";s:4:"7e30";s:45:"resources/jscalendar/lang/calendar-sk-utf8.js";s:4:"48af";s:40:"resources/jscalendar/lang/calendar-sk.js";s:4:"1565";s:40:"resources/jscalendar/lang/calendar-sp.js";s:4:"c97f";s:45:"resources/jscalendar/lang/calendar-sv-utf8.js";s:4:"5496";s:40:"resources/jscalendar/lang/calendar-sv.js";s:4:"c1cd";s:40:"resources/jscalendar/lang/calendar-tr.js";s:4:"b692";s:40:"resources/jscalendar/lang/calendar-zh.js";s:4:"9a07";s:56:"resources/naturalLanguageParser/naturalLanguageParser.js";s:4:"2ab9";s:46:"resources/naturalLanguageParser/patterns/de.js";s:4:"9a24";s:46:"resources/naturalLanguageParser/patterns/en.js";s:4:"62d6";s:44:"resources/naturalLanguageParser/help/de.html";s:4:"bf4e";s:44:"resources/naturalLanguageParser/help/en.html";s:4:"2493";s:14:"doc/manual.sxw";s:4:"3834";s:24:"src/class.jscalendar.php";s:4:"0000";s:32:"src/class.tx_date2cal_befunc.php";s:4:"94e4";s:32:"src/class.tx_date2cal_shared.php";s:4:"e3ba";s:32:"src/class.tx_date2cal_wizard.php";s:4:"77dd";s:29:"src/class.ux_sc_db_layout.php";s:4:"f7c7";}',
);

?>